.circleci/config.yml

```diff
@@ -73,8 +88,8 @@ jobs:
 
       - save_cache:
           paths:
-            - ~/.ccache
-          key: -{{ checksum "Makefile.envs" }}-v20201205-{{ .BuildNum }}
+            - /root/.ccache
+          key: -core-{{ checksum "Makefile.envs" }}-v20210910-
```

Comment:
No longer store the cache for each build, as then we are not sure which one gets used, and it tend to grows with each build if there are cache misses

Reply:
So is the assumption here that the ccache will only be updated if `Makefile.envs` changes? Maybe it would be better to key also on `Makefile` and `cpython/Makefile`?

Reply:
Yes, it's a balance minimizing cache resets and doing that if something major changes. Currently `Makefile.envs` includes all the global flags. 
If I look at git history,
 - `Makefile` most of the changes there are either about js packaging, linting or something other about build that doesn't affect compilation
 - `cpython/Makefile`: most changes also also rather local (even libffi addition). Also I'd like to move most of those under the main build system as standalone packages eventually if possible.

The cost of rebuilding without cache is roughly 40min, and the issue is that the cache for each fork is independent as far as I can tell, so it's multiplied by the number of contributors. So I think even if a few compilation options change for a subset of files in a package (including CPython), and we get some cache misses overall it's probably still fine to keep the same cache. With the assumption that we update emsdk periodically which resets the cache anyway.

Reply:
Actually, yes for the core build we can certainly include `cpython/Makefile` in the hash since the core/package caches are separate.